writers/local/local.go

```diff
@@ -34,35 +36,38 @@ type Local struct {
 	writer              *goparquet.FileWriter
 	stream              protocol.Stream
 	records             atomic.Int64
-	pqSchemaMutex       sync.Mutex // To prevent concurrent underlying map access from fraugster library
+	pqSchemaMutex       sync.Mutex // To prevent concurrent map access from fraugster library
+	s3Client            *s3.S3
+	s3KeyPath           string
 }
 
+// GetConfigRef returns the config reference for the local writer.
 func (l *Local) GetConfigRef() protocol.Config {
 	l.config = &Config{}
 	return l.config
 }
 
+// Spec returns a new Config instance.
 func (l *Local) Spec() any {
 	return Config{}
 }
 
+// Setup configures the local writer, including paths, file names, and optional S3 setup.
 func (l *Local) Setup(stream protocol.Stream, options *protocol.Options) error {
 	l.options = options
 	l.fileName = utils.TimestampedFileName(constants.ParquetFileExt)
-	l.destinationFilePath = filepath.Join(l.config.BaseFilePath, stream.Namespace(), stream.Name(), l.fileName)
+	l.destinationFilePath = filepath.Join(l.config.Path, stream.Namespace(), stream.Name(), l.fileName)
 
-	// Start a new local writer
-	err := os.MkdirAll(filepath.Dir(l.destinationFilePath), os.ModePerm)
-	if err != nil {
-		return err
+	// Create directories
+	if err := os.MkdirAll(filepath.Dir(l.destinationFilePath), os.ModePerm); err != nil {
+		return fmt.Errorf("failed to create directories: %w", err)
 	}
 
+	// Initialize local Parquet writer
 	pqFile, err := local.NewLocalFileWriter(l.destinationFilePath)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create local file writer: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("failed to setup parquet file writer: %w", err)
```

use parquet in place of local